org.scala.tools.eclipse.search/src/org/scala/tools/eclipse/search/indexing/Index.scala

```diff
@@ -79,6 +81,15 @@ trait Index extends HasLogger {
     location(project).toFile.exists
   }
 
+  def indexIsClean(project: IProject): Boolean = {
```

Comment:
indexIsClean is not a good name for a side effecting method. They should always start with a verb. I would use `cleanIndex`. Also, it should return a `Try[Boolean]`, since you want to be able to do something on an error (or is `using` already doing some error handling?).